src/components/form/date-input/date-input.component.ts

```diff
@@ -18,38 +18,66 @@ import { ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR, Validator } fro
     }
   ],
 })
-export class DateInputComponent implements ControlValueAccessor, Validator, OnDestroy {
+export class DateInputComponent implements ControlValueAccessor, Validator, OnInit, OnDestroy {
   @Input()
   public id: string;
 
   @Input()
   public mandatory: boolean;
 
+  @Input()
+  public dateTime: boolean;
+
   @Input()
   public formControl: FormControl;
 
   public displayDay: string = null;
   public displayMonth: string = null;
   public displayYear: string = null;
 
+  public displayHour: string = null;
+  public displayMinute: string = null;
+  public displaySecond: string = null;
   // tslint:disable-next-line
-  private readonly DATE_FORMAT = /^(\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/;
-
+  // private readonly DATE_FORMAT = /^(\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/;
+  private readonly DATE_FORMAT =
+    /^(\d{4})-?(\d\d)-?(\d\d)(?:T(\d?\d)(?::?(\d?\d)(?::?(\d\d)(?:\.(\d+))?)?)?(Z|([+-])(\d\d):?(\d\d))?|Z)?$/;
```

Comment:
to make this simpler to understand and to maintain, can we break it into subpatterns? I've found this nice example on stackoverflow:

`var box1 = "([^<>()[\]\\\\.,;:\s@\"]+(\\.[^<>()[\\]\\\\.,;:\s@\"]+)*)";
var box2 = "(\".+\")";
var host1 = "(\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\])";
var host2 = "(([a-zA-Z\-0-9]+\\.)+[a-zA-Z]{2,})";

var regex = new RexExp("^(" + box1 + "|" + box2 + ")@(" + host1 + "|" + host2 + ")$");`